internal/databases/postgres/bguploader.go

```diff
@@ -220,7 +220,8 @@ func (b *BgUploader) upload(walStatusFilename string) bool {
 		return false
 	}
 
-	if err := b.uploader.ArchiveStatusManager.MarkWalUploaded(walFilename); err != nil {
+	err = b.uploader.ArchiveStatusManager.MarkWalUploaded(walFilename)
```

Comment:
Revert this. You're overwriting the error from `uploadWALFile`

Reply:
Oh I see, `uploadWALFile` has return false, so this makes sense. So if this is breaking a test the test is wrong